worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -534,49 +525,49 @@ private[celeborn] class Worker(
 
   private def handleResourceConsumption(): util.Map[UserIdentifier, ResourceConsumption] = {
     val resourceConsumptionSnapshot = storageManager.userResourceConsumptionSnapshot()
-    resourceConsumptionSnapshot.foreach { resourceConsumption =>
-      {
+    resourceConsumptionSnapshot.foreach { case (userIdentifier, _) =>
+      if (!workerInfo.userResourceConsumption.containsKey(userIdentifier)) {
         resourceConsumptionSource.addGauge(
           ResourceConsumptionSource.DISK_FILE_COUNT,
-          resourceConsumption._1.toMap) { () =>
-          computeUserResourceConsumption(resourceConsumption).diskFileCount
+          userIdentifier.toMap) { () =>
+          if (workerInfo.userResourceConsumption.containsKey(userIdentifier)) {
+            workerInfo.userResourceConsumption.get(userIdentifier).diskFileCount
+          } else {
+            0L
+          }
         }
         resourceConsumptionSource.addGauge(
           ResourceConsumptionSource.DISK_BYTES_WRITTEN,
-          resourceConsumption._1.toMap) { () =>
-          computeUserResourceConsumption(resourceConsumption).diskBytesWritten
+          userIdentifier.toMap) { () =>
+          if (workerInfo.userResourceConsumption.containsKey(userIdentifier)) {
+            workerInfo.userResourceConsumption.get(userIdentifier).diskBytesWritten
+          } else {
+            0L
+          }
         }
         resourceConsumptionSource.addGauge(
           ResourceConsumptionSource.HDFS_FILE_COUNT,
-          resourceConsumption._1.toMap) { () =>
-          computeUserResourceConsumption(resourceConsumption).hdfsFileCount
+          userIdentifier.toMap) { () =>
+          if (workerInfo.userResourceConsumption.containsKey(userIdentifier)) {
+            workerInfo.userResourceConsumption.get(userIdentifier).hdfsFileCount
+          } else {
+            0L
+          }
         }
         resourceConsumptionSource.addGauge(
           ResourceConsumptionSource.HDFS_BYTES_WRITTEN,
-          resourceConsumption._1.toMap) { () =>
-          computeUserResourceConsumption(resourceConsumption).hdfsBytesWritten
+          userIdentifier.toMap) { () =>
+          if (workerInfo.userResourceConsumption.containsKey(userIdentifier)) {
+            workerInfo.userResourceConsumption.get(userIdentifier).hdfsBytesWritten
+          } else {
+            0L
+          }
         }
       }
     }
     workerInfo.updateThenGetUserResourceConsumption(resourceConsumptionSnapshot.asJava)
```

Comment:
when workerInfo updateThenGetUserResourceConsumption,  it will first clear userResourceConsumption, then it may cause user resource consumption metrics to 0.

Reply:
> when workerInfo updateThenGetUserResourceConsumption, it will first clear userResourceConsumption, then it may cause user resource consumption metrics to 0.

Nice catch, it's a problem.

Reply:
@RexXiong How about current?

Reply:
How about current?